lib/json-schema/attributes/ref.rb

```diff
@@ -21,21 +22,23 @@ def self.validate(current_schema, data, fragments, processor, validator, options
       def self.get_referenced_uri_and_schema(s, current_schema, validator)
         uri,schema = nil,nil
 
-        temp_uri = Addressable::URI.parse(s['$ref'])
-        if temp_uri.relative?
-          temp_uri = current_schema.uri.clone
-          # Check for absolute path
-          path = s['$ref'].split("#")[0]
-          if path.nil? || path == ''
-            temp_uri.path = current_schema.uri.path
-          elsif path[0,1] == "/"
-            temp_uri.path = Pathname.new(path).cleanpath.to_s
-          else
-            temp_uri = current_schema.uri.join(path)
+        temp_uri = JSON::Util::URI.parse(s['$ref'])
+        temp_uri.defer_validation do
+          if temp_uri.relative?
+            temp_uri.merge!(current_schema.uri)
+            # Check for absolute path
+            path, fragment = *s['$ref'].split("#")
```

Comment:
Interesting - I always thought that was necessary for array restructuring (but apparently only if one of the destination variables should be an array)